templates/search_clients_daily_v7.sql

```diff
@@ -0,0 +1,156 @@
+      ARRAY(
+      SELECT
+        AS STRUCT element.source AS source,
+        element.engine AS engine,
+        element.count AS count,
+        CASE
+          WHEN (element.source IN ('searchbar',  'urlbar',  'abouthome',  'newtab',  'contextmenu',  'system',  'activitystream',  'webextension',  'alias') OR element.source IS NULL) THEN 'sap'
+          WHEN (STARTS_WITH(element.source, 'in-content:sap:')
+          OR STARTS_WITH(element.source, 'sap:')) THEN 'tagged-sap'
+          WHEN (STARTS_WITH(element.source, 'in-content:sap-follow-on:') OR STARTS_WITH(element.source,'follow-on:')) THEN 'tagged-follow-on'
+          WHEN STARTS_WITH(element.source, 'in-content:organic:') THEN 'organic'
+          WHEN STARTS_WITH(element.source, 'ad-click:') THEN 'ad-click'
+          WHEN STARTS_WITH(element.source, 'search-with-ads:') THEN 'search-with-ads'
+        ELSE
+        'unknown'
+      END
+        AS type
+      FROM
+        UNNEST(search_counts.list) ),
+      ARRAY(
+      SELECT
+        AS STRUCT "ad-click:" AS source,
+        key AS engine,
+        value AS count,
+        "ad-click" AS type
+      FROM
+        UNNEST(scalar_parent_browser_search_ad_clicks.key_value) ),
+      ARRAY(
+      SELECT
+        AS STRUCT "search-with-ads:" AS source,
+        key AS engine,
+        value AS count,
+        "search-with-ads" AS type
+      FROM
+        UNNEST(scalar_parent_browser_search_with_ads.key_value) ) ) AS _searches
+  FROM
+    main_summary_v4 ),
+  flattened AS (
+  SELECT
+    *
+  FROM
+    augmented,
+    UNNEST(
+    IF
+      -- Provide replacement empty _searches with one null search, to ensure all
+      -- clients are included in results
+      (ARRAY_LENGTH(_searches) > 0,
+        _searches,
+        [(CAST(NULL AS STRING),
+          CAST(NULL AS STRING),
+          NULL,
+          CAST(NULL AS STRING))])) ),
+  -- Aggregate by client_id using windows
+  windowed AS (
+  SELECT
+    ROW_NUMBER() OVER w1_unframed AS _n,
+    submission_date_s3 AS submission_date,
+    client_id,
+    engine,
+    source,
+    FIRST_VALUE(country) OVER w1 AS country,
+    FIRST_VALUE(app_version) OVER w1 AS app_version,
+    FIRST_VALUE(distribution_id) OVER w1 AS distribution_id,
+    FIRST_VALUE(locale) OVER w1 AS locale,
+    FIRST_VALUE(user_pref_browser_search_region) OVER w1 AS user_pref_browser_search_region,
+    FIRST_VALUE(search_cohort) OVER w1 AS search_cohort,
+    FIRST_VALUE(os) OVER w1 AS os,
+    FIRST_VALUE(os_version) OVER w1 AS os_version,
+    FIRST_VALUE(channel) OVER w1 AS channel,
+    FIRST_VALUE(profile_creation_date) OVER w1 AS profile_creation_date,
+    FIRST_VALUE(default_search_engine) OVER w1 AS default_search_engine,
+    FIRST_VALUE(default_search_engine_data_load_path) OVER w1 AS default_search_engine_data_load_path,
+    FIRST_VALUE(default_search_engine_data_submission_url) OVER w1 AS default_search_engine_data_submission_url,
+    FIRST_VALUE(sample_id) OVER w1 AS sample_id,
```

Comment:
That makes sense to me, it seems like we're just using that function to get a column for the aggregation. ++ on being consistent with `clients_daily`

Probably best to get @harterrt's or @benmiroglio's opinion on this since they've actually used this dataset for analysis. :)

Reply:
+1 to being deterministic here.